Branch-SDK/src/io/branch/indexing/ContentDiscoverer.java

```diff
@@ -165,7 +165,12 @@ public void run() {
 
                         // Cache the analytics data for future use
                         PrefHelper.getInstance(activity).saveBranchAnalyticsData(contentEvent_);
-                        lastActivityReference_ = null;
+
+                        if (cdManifest_.getDiscoveryRepeatTime() != -1) {
```

Comment:
as in
private static final int DRT_NOT_SET = -1;
and then if getDiscoveryRepeatTime() != DRT_NOT_SET
?